### ISSUE TITLE
Building with MSYS2 / Clang++

### ISSUE THREAD

**Issue by dresch86** [mosra/magnum#227]
I managed to get Magnum to build with the following command:

```
cmake .. -G "MSYS Makefiles" -DCMAKE_INSTALL_PREFIX="C:/Libs/magnum" -DCMAKE_PREFIX_PATH="C:/Libs/corrade/bin;C:/Libs/corrade" -DCMAKE_INSTALL_RPATH="lib64/" -DWITH_SDL2APPLICATION=ON -DWITH_DEBUGTOOLS=OFF -DLIB_SUFFIX=64 -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXX_FLAGS="-fno-rtti"
```

Adding the -fno-rtti was necessary otherwise the linker complained about 'undefined reference to `typeinfo for Magnum::AbstractShaderProgram'. Just thought I would throw that out there in case anyone else was wondering.... When I get a chance to learn the inner working of CMake I will make this less verbose :-)

**Comment by mosra** [comment]
Amazing, thank you!

Can you tell from where the undefined reference error was coming (during linking of which library)?

**Comment by primesoftware** [comment]
> Amazing, thank you!

Can you tell from where the undefined reference error was coming (during linking of which library)?

Yes, it was the Shaders library.

**Comment by mosra** [comment]
As of f6abe4577cf9e5e4faa9c7de5803a9fbb8d7b856 Magnum supports clang-cl on Windows and various Clang-specific linker errors on Windows were fixed. Since this issue is almost two years old and there was a lot of changes happening since, I'm not sure if this particular linker issue still applies.

**Comment by mosra** [comment]
Work to fix linker errors is being done in #417, but the resulting build seems to be crashing in weird places (`UtilityFatalTest`, all GL/AL tests) with no apparent reason. Reopening and putting this on hold for now.

**Comment by mosra** [comment]
As far as I can tell, the remaining issues were handled with #439, closing.
